sensor_filter.go

```diff
@@ -0,0 +1,168 @@
+	// model-specific (aka "native") Config structure defined above, making it easier to directly access attributes.
+	conf, err := resource.NativeConfig[*Config](rawConf)
+	if err != nil {
+		return nil, err
+	}
+
+	// Create a cancelable context
+	cancelCtx, cancelFunc := context.WithCancel(context.Background())
+
+	s := &sensorFilter{
+		name:       rawConf.ResourceName(),
+		logger:     logger,
+		cfg:        conf,
+		cancelCtx:  cancelCtx,
+		cancelFunc: cancelFunc,
+	}
+
+	// The Reconfigure() method changes the values on the service based on the attributes in the component config
+	if err := s.Reconfigure(ctx, deps, rawConf); err != nil {
+		logger.Error("Error configuring module with ", rawConf)
+		return nil, err
+	}
+
+	return s, nil
+}
+
+// Methods in this service will be on this struct
+type sensorFilter struct {
+	name   resource.Name
+	logger logging.Logger
+	cfg    *Config
+
+	cancelCtx  context.Context
+	cancelFunc func()
+
+	sensorName string
+	sensor     sensor.Sensor
+	reading    string
+	condition  []*apputils.Eval
+}
+
+func (s *sensorFilter) Name() resource.Name {
+	return s.name
+}
+
+// Reconfigures the model. Most models can be reconfigured in place without needing to rebuild. If you need to instead create a new instance of the servo, throw a NewMustBuildError.
+func (s *sensorFilter) Reconfigure(ctx context.Context, deps resource.Dependencies, conf resource.Config) error {
+	sfConfig, err := resource.NativeConfig[*Config](conf)
+	if err != nil {
+		s.logger.Warn("Error reconfiguring module with ", err)
+		return err
+	}
+	s.name = conf.ResourceName()
+
+	// Make sure we can get the sensor
+	s.sensorName = sfConfig.SensorName
+	s.sensor, err = sensor.FromDependencies(deps, sfConfig.SensorName)
+	if err != nil {
+		return errors.Wrapf(err, "could not find sensor called %v ", s.sensorName)
+	}
+
+	// Get everything else
+	s.condition = sfConfig.Conditions
+	if len(s.condition) < 1 {
+		s.logger.Warn("Condition(s) not found. Without conditions, the result will always be true.")
+	}
+	s.reading = sfConfig.Reading
+
+	return nil
+}
+
+// The DoCommand for this module will evaluate a sensor reading against EVERY condition set up in the
+// service config. The "result" will be true if and only if all conditions are met. Otherwise, it's false.
+func (s *sensorFilter) DoCommand(ctx context.Context, cmd map[string]interface{}) (map[string]interface{}, error) {
+	// Read from sensor
+	sensorOut, err := s.sensor.Readings(ctx, nil)
+	if err != nil {
+		return nil, errors.Wrapf(err, "could not get readings from sensor %v", s.sensorName)
+	}
+
+	// Evaluate sensor reading against each condition
+	// We want the logical AND of them all
+	finalAns := true
+	for _, eval := range s.condition {
+		ans, err := eval.Operator.Evaluate(sensorOut[s.reading], eval.Value)
+		if err != nil {
+			s.logger.Error(err)
+			return nil, err
+		}
+		finalAns = finalAns && ans
+	}
```

Comment:
Since you do an AND, when you first encounter a false, you can stop early and just return a false